Source/ExcelDna.Integration/ExcelSynchronizationContext.cs

```diff
@@ -363,8 +363,20 @@ public bool IsRegistered
         void Unregister()
         {
             // Clear the name and unregister
-            XlCall.Excel(XlCall.xlfSetName, _syncMacroName);
-            XlCall.Excel(XlCall.xlfUnregister, _syncMacroRegistrationId);
+            try
```

Comment:
These calls should rather be XlCall>TryExcel.

Reply:
I've now found that if I open Excel without opening a workbook, with ExcelDna.Test add-in, it fails on closing Excel with AccessViolationException on the Excel12v call line at 

> 	ExcelDna.Loader!ExcelDna.Loader.XlCallImpl.TryExcelImpl12(int xlFunction, out object result, object[] parameters) Line 123	C#
>  	ExcelDna.Loader!ExcelDna.Loader.XlCallImpl.TryExcelImpl(int xlFunction, out object result, object[] parameters) Line 75	C#
>  	ExcelDna.Loader!ExcelDna.Loader.IntegrationHost.TryExcelImpl(int xlFunction, out object result, object[] parameters) Line 31	C#
>  	ExcelDna.Integration.dll!ExcelDna.Integration.ExcelIntegration.TryExcelImpl(int xlFunction, out object result, object[] parameters) Line 32	C#
>  	ExcelDna.Integration.dll!ExcelDna.Integration.XlCall.TryExcel(int xlFunction, out object result, object[] parameters) Line 1111	C#
>  	ExcelDna.Integration.dll!ExcelDna.Integration.XlCall.Excel(int xlFunction, object[] parameters) Line 1094	C#
>  	ExcelDna.Integration.dll!ExcelDna.Integration.RunMacroSynchronization.Unregister() Line 368	C#
>  	ExcelDna.Integration.dll!ExcelDna.Integration.RunMacroSynchronization.Dispose() Line 385	C#
>  	ExcelDna.Integration.dll!ExcelDna.Integration.SynchronizationWindow.Dispose() Line 729	C#
>  	ExcelDna.Integration.dll!ExcelDna.Integration.SynchronizationManager.Uninstall() Line 66	C#
>    ExcelDna.Integration.dll!ExcelDna.Integration.ExcelComAddIn.OnDisconnection(ExcelDna.Integration.Extensibility.ext_DisconnectMode RemoveMode, ref System.Array custom) Line 50	C#

So, on shutdown we probably shouldn't call Excel at all.